apps/api_web/lib/api_web/api_controller_helpers.ex

```diff
@@ -161,11 +164,16 @@ defmodule ApiWeb.ApiControllerHelpers do
 
         fields
         |> String.split(",")
-        |> Enum.filter(&MapSet.member?(attribute_set, &1))
+        |> Enum.filter(&(MapSet.member?(attribute_set, &1) || older_param(conn, type, &1)))
         |> Enum.map(&String.to_existing_atom/1)
     end
   end
 
+  def older_param(%{assigns: %{api_version: ver}}, "facility", "name") when ver < "2019-07-01",
```

Comment:
Rather than having this as an addition to `ApiControllerHelpers`, can you defer to the `view_module`? Something like:
```elixir
attribute_set = view_module.attribute_set(conn)

# in api_view_helpers
def attribute_set(_conn) do
  MapSet.new(__MODULE__.__attributes(), &Atom.to_string/1)
end
defoverridable [attribute_set: 1]

# in FacilityView
def attribute_set(conn) do
  ...behavior...
end
```

Reply:
Yes, it looks much better now, thanks. Refactored.